pkg/privilege/privileges/cache.go

```diff
@@ -521,147 +527,143 @@ func findUserAndAllRoles(all map[string]struct{}, roleGraph map[string]roleGraph
 			break
 		}
 	}
+	return all
 }
 
-func (p *MySQLPrivilege) loadSomeUsers(ctx sqlexec.RestrictedSQLExecutor, userList ...string) ([]string, error) {
-	if len(userList) > 512 {
-		logutil.BgLogger().Warn("loadSomeUsers called with a long user list", zap.Int("len", len(userList)))
```

Comment:
This limitation is eliminated, by load all data and do the filter condition in the user space.